src/cephadm/cephadm

```diff
@@ -936,6 +936,38 @@ def infer_fsid(func):
 
     return _infer_fsid
 
+def infer_image(func):
+    """
+    Use the most recent ceph image
+    """
+    @wraps(func)
+    def _infer_image():
+        if not args.image:
+            args.image = get_last_local_ceph_image()
+        if not args.image:
+            args.image = DEFAULT_IMAGE
+        if not args.image:
+            args.image = os.environ.get('CEPHADM_IMAGE')
```

Comment:
@liewegas some notes here:
1) this is dead code because at this point `args.image` will always have value (`DEFAULT_IMAGE` value, at least)
2) If it's intentional to use `CEPHADM_IMAGE` env var value after `get_last_local_ceph_image` value, this means that it's not possible to use `CEPHADM_IMAGE` env to specify the image when we already have an image pulled (env var will be ignored). In this case, I think the correct name for this env var would be `DEFAULT_CEPHADM_IMAGE`. I understand if we cannot rename it at this point, but at least we should fix the `--image` documentation:
```
parser.add_argument(
        '--image',
        help='container image. Can also be set via the "CEPHADM_IMAGE" '
        'env var')
```

Reply:
fixed by moving the env check above get_last_local_ceph_image().